src/python/pants/engine/intrinsics.py

```diff
@@ -120,13 +121,39 @@ async def run_id() -> RunId:
     return await native_engine.run_id()
 
 
+# NB: Call one of the helpers below, instead of calling this rule directly,
+#  to ensure correct application of restartable logic.
 @_uncacheable_rule
-async def interactive_process(
+async def _interactive_process(
     process: InteractiveProcess, process_execution_environment: ProcessExecutionEnvironment
 ) -> InteractiveProcessResult:
     return await native_engine.interactive_process(process, process_execution_environment)
 
 
+async def run_interactive_process(iproc: InteractiveProcess) -> InteractiveProcessResult:
+    # NB: We must call mark_nonrestartable() in this helper, rather than in a nested @rule call,
```

Comment:
Question: why don't we need to do both? AIUI, there's two tasks here, the parent that's running this code, a child that's running the Rust intrinsic. It seems like marking the parent restartable might still risk the child restarting?

Reply:
I'm assuming that it's not necessary in the same way that it's not necessary for all the other @rules invoked transitively by a goal... But @stuhood's input will be important here.